GeneratorInterface/RivetInterface/plugins/RivetAnalyzer.cc

```diff
@@ -119,7 +116,7 @@ void RivetAnalyzer::analyze(const edm::Event& iEvent,const edm::EventSetup& iSet
       if(!_useLHEweights){
 	edm::Handle<GenEventInfoProduct> genEventInfoProduct;
 	iEvent.getByToken(_genEventInfoCollection, genEventInfoProduct);
-	tmpGenEvtPtr->weights()[0] = genEventInfoProduct->weight();
+	tmpGenEvtPtr->weights()[0] = genEventInfoProduct->weights().at(_LHEweightNumber);
```

Comment:
isn't "LHE" a bit misleading here?
ps what is the use case? weights from pythia?